pkg/zonemgr/zone_file_test.go

```diff
@@ -42,15 +43,18 @@ var _ = Describe("disk zone file maintenance", func() {
 	})
 
 	When("zone file already exist", func() {
-		BeforeEach(func() {
-			Expect(os.WriteFile(zoneFileName, []byte(zoneFileContent), zoneFilePerm)).To(Succeed())
-		})
 		AfterEach(func() {
 			Expect(os.RemoveAll(zoneFileName)).To(Succeed())
 		})
 
 		It("should override a zone file on disk", func() {
+			Expect(os.WriteFile(zoneFileName, []byte(zoneFileContent), zoneFilePerm)).To(Succeed())
 			testFileFunc(zoneFileUpdatedContent)
 		})
+
+		It("should read SOA serial", func() {
```

Comment:
Please add a test testing `ReadSoaSerial` to the scenario the file doesn't exist.

Reply:
done